AsyncDisplayKit/ASVideoPlayerNode.h

```diff
@@ -0,0 +1,147 @@
+
+/**
+ * @abstract Delegate method invoked in layoutSpecThatFits:
+ * @param videoPlayer
+ * @param controls - Dictionary of controls which are used in videoPlayer; Dictionary keys are ASVideoPlayerNodeControlType
+ * @param maxSize - Maximum size for ASVideoPlayerNode
+ * @discussion - Developer can layout whole ASVideoPlayerNode as he wants. ASVideoNode is locked and it can't be changed
+ */
+- (ASLayoutSpec *)videoPlayerNodeLayoutSpec:(ASVideoPlayerNode *)videoPlayer
+                                forControls:(NSDictionary *)controls
+                             forMaximumSize:(CGSize)maxSize;
+
+#pragma mark Text delegate methods
+/**
+ * @abstract Delegate method invoked before creating ASVideoPlayerNodeControlTypeElapsedText and ASVideoPlayerNodeControlTypeDurationText
+ * @param videoPlayer
+ * @param timeLabelType
+ */
+- (NSDictionary *)videoPlayerNodeTimeLabelAttributes:(ASVideoPlayerNode *)videoPlayerNode timeLabelType:(ASVideoPlayerNodeControlType)timeLabelType;
+- (NSString *)videoPlayerNode:(ASVideoPlayerNode *)videoPlayerNode
+   timeStringForTimeLabelType:(ASVideoPlayerNodeControlType)timeLabelType
+                      forTime:(CMTime)time;
+
+#pragma mark Scrubber delegate methods
+- (UIColor *)videoPlayerNodeScrubberMaximumTrackTint:(ASVideoPlayerNode *)videoPlayer;
+- (UIColor *)videoPlayerNodeScrubberMinimumTrackTint:(ASVideoPlayerNode *)videoPlayer;
+- (UIColor *)videoPlayerNodeScrubberThumbTint:(ASVideoPlayerNode *)videoPlayer;
+- (UIImage *)videoPlayerNodeScrubberThumbImage:(ASVideoPlayerNode *)videoPlayer;
+
+#pragma mark - Spinner delegate methods
+- (UIColor *)videoPlayerNodeSpinnerTint:(ASVideoPlayerNode *)videoPlayer;
+
+#pragma mark - Playback button delegate methods
+- (UIColor *)videoPlayerNodePlaybackButtonTint:(ASVideoPlayerNode *)videoPlayer;
+
+
+#pragma mark ASVideoNodeDelegate proxy methods
+/**
+ * @abstract Delegate method invoked when ASVideoPlayerNode playback time is taped.
+ * @param videoPlayerNode The ASVideoPlayerNode that was tapped.
+ */
+- (void)didTapVideoPlayerNode:(ASVideoPlayerNode *)videoPlayer;
+/**
+ * @abstract Delegate method invoked when ASVideoNode playback time is updated.
+ * @param videoPlayerNode The video player node
+ * @param second current playback time.
+ */
+- (void)videoPlayerNode:(ASVideoPlayerNode *)videoPlayer didPlayToTime:(CMTime)time;
+
+/**
+ * @abstract Delegate method invoked when ASVideoNode changes state.
+ * @param videoPlayerNode The ASVideoPlayerNode whose ASVideoNode is changing state.
+ * @param state ASVideoNode state before this change.
+ * @param toSate ASVideoNode new state.
+ * @discussion This method is called after each state change
+ */
+- (void)videoPlayerNode:(ASVideoPlayerNode *)videoPlayer willChangeVideoNodeState:(ASVideoNodePlayerState)state toVideoNodeState:(ASVideoNodePlayerState)toState;
+
+/**
+ * @abstract Delegate method is invoked when ASVideoNode decides to change state.
```

Comment:
Need more explanation of what this method does (aka not allowing certain state change..etc)